docs/user-guide/command-line-interface.md

```diff
@@ -122,18 +122,22 @@ Examples:
 
 #### `--ext`
 
-This option allows you to specify which file extensions ESLint will use when searching for JavaScript files. By default, it uses `.js` as the only file extension.
+This option allows you to specify which file extensions ESLint will use when searching for JavaScript files in the directories you specify.
+By default, it uses `.js` as the only file extension.
 
 Examples:
 
     # Use only .js2 extension
-    eslint --ext .js2
+    eslint . --ext .js2
 
     # Use both .js and .js2
-    eslint --ext .js --ext .js2
+    eslint . --ext .js --ext .js2
 
     # Also use both .js and .js2
-    eslint --ext .js,.js2
+    eslint . --ext .js,.js2
+
+**Note:** This option has no effect on glob parameters.
+For example, `eslint lib/* --ext .js` will match all files within the `lib/` directory, regardless of extension.
```

Comment:
A glob is something entirely different from specifying a filename or a directory name.  A glob is a pattern containing a wildcard (such as `*` or `**`).  In fact, I am showing an unhappy path (sad path?) with `eslint lib/* --ext .js`, as in this case the `--ext` has no effect because a glob was used, whereas it would work fine as `eslint lib --ext .js` (linting a directory name, not a glob).  

Or, maybe I'm still misunderstanding what you're saying?  All flags assume that a pattern (filename, directory, or glob) was specified, I think it would be tedious to mention that on all of them, and confusing to mention it only for one.